Source/WebCore/page/LocalDOMWindow.cpp

```diff
@@ -2408,11 +2408,6 @@ void LocalDOMWindow::removeAllEventListeners()
     m_touchAndGestureEventListenerCount = 0;
 #endif
 
-#if ENABLE(TOUCH_EVENTS)
-    if (RefPtr document = this->document())
```

Comment:
The less risky change would be to keep the code as-is but switch from `RefPtr` to `RefPtrAllowingPartiallyDestroyed<Document>`.

Your new code is not equivalent since `Document::removeAllEventListeners()` is NOT the only caller of `LocalDOMWindow::removeAllEventListeners()`.

Reply:
Didn't know about `RefPtrAllowingPartiallyDestroyed`. Thank you for your advice.